### PR TITLE
Fix typos in docs

### PR DESCRIPTION
**Pull request by tocic**
Use "Acknowledgments" everywhere for consistency.

### CODE REVIEW

**Comment by Lastique** [comment]
"Acknowledgement" is the [correct](https://dictionary.cambridge.org/dictionary/english/acknowledgement) spelling.

File: doc/scoped_enum.qbk [hunk 1]
```diff
@@ -46,7 +46,7 @@ The user can portably declare such enumeration as follows:
     }
     BOOST_SCOPED_ENUM_DECLARE_END(future_errc)
 
-These macros allows to use `future_errc` in almost all the cases as an scoped enum.
+These macros allow using `future_errc` in almost all the cases as a scoped enum.
```
Comment (Lastique) [inline doc/scoped_enum.qbk:49]: "Allow to use" seems correct to me. Is it not? I'm not a native speaker.

Comment (tocic) [reply doc/scoped_enum.qbk:49]: Please see https://english.stackexchange.com/q/60271.

**Comment by tocic** [comment]
> "Acknowledgement" is the [correct](https://dictionary.cambridge.org/dictionary/english/acknowledgement) spelling.

See https://english.stackexchange.com/q/390960. I chose the "Acknowledgments" version, because [searching through the repo for it](https://github.com/boostorg/core/search?q=%22Acknowledgments%22) yields 6 results versus 2 for "Acknowledgements". I don't insist on this spelling, but it's better to be consistent. I'll convert everything to "Acknowledgements" if you wish.

**Comment by Lastique** [comment]
On September 18, 2022 4:45:30 PM tocic ***@***.***> wrote:

>> "Acknowledgement" is the 
>> [correct](https://dictionary.cambridge.org/dictionary/english/acknowledgement) 
>> spelling.
>
> See https://english.stackexchange.com/q/390960. I chose the 
> "Acknowledgments" version, because [searching through the repo for 
> it](https://github.com/boostorg/core/search?q=%22Acknowledgments%22) yields 
> 6 results versus 2 for "Acknowledgements". I don't insist on this spelling, 
> but it's better to be consistent. I'll convert everything to 
> "Acknowledgements" if you wish.

My personal preference would be to keep the 'e', mostly because that's how 
I'm used to. Other maintainers and authors may have different preferences. 
If both spellings are correct then I don't mind keeping every instance as 
is. But if you want to convert everything to one spelling (with 'e') , I 
won't object either.

**Review by Lastique — APPROVED** [review]

**Pull request merged**